libpod/runtime_img.go

```diff
@@ -96,30 +173,35 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 
 	// PullImage copies the image from the source to the destination
 	var (
-		images []string
+		images       []string
+		fromRegistry bool
 	)
 
-	if signaturePolicyPath == "" {
-		signaturePolicyPath = r.config.SignaturePolicyPath
+	signaturePolicyPath := r.config.SignaturePolicyPath
+	if options.SignaturePolicyPath != "" {
+		signaturePolicyPath = options.SignaturePolicyPath
 	}
 
-	sc := common.GetSystemContext(signaturePolicyPath, "")
+	sc := common.GetSystemContext(signaturePolicyPath, options.AuthFile)
```

Comment:
This `sc` is primarily used for `signature.DefaultPolicy`, which does not use `AuthFile`.  `AuthFile` needs to be set in `copy.Options.{Source,Destination}Ctx`, and that is not happening AFAICS.

Reply:
Right, I thought setting it in systecmcontext is what we needed. Surprisingly the code reaches the autfile and gets the cached credentials from there.
I think because the policyContext is set using the authfile, which is passed on to cp.Image so it is able to access the credentials from the authfile?

Reply:
> Right, I thought setting it in systecmcontext is what we needed.

Yes, but there isn’t a per-process `SystemContext`. Each function can take its own configuration.

> Surprisingly the code reaches the autfile and gets the cached credentials from there.
I think because the policyContext is set using the authfile, which is passed on to cp.Image so it is able to access the credentials from the authfile?

No, ultimately `PolicyContext` does not carry a pointer to the `SystemContext` at all, so `copy.Image` can not use it even if it tried. The data flow should be `copy.Image:{srcRef.NewImageSource(options.SourceCtx),destRef.NewImageDestination(options.DestinationCtx)}` → `dockerReference.NewImage{Source,Destination}` → `docker.newImage{Source,Destination}` → `docker.newDockerClientFromRef` → `config.GetAuthentication`.

Are you trying a non-default authfile path?  The default file in `$XDG_RUNTIME_DIR` is used without having to set any option, that’s the only explanation I can think of.

Reply:
You are right. I was trying a non-default authfile but was logged in with docker under root and didn't realize that is where it was getting my credentials from.
Making the fix. Thanks!

Reply:
Fixed.